cmake/app/boilerplate.cmake

```diff
@@ -180,7 +180,11 @@ set(CACHED_BOARD ${BOARD} CACHE STRING "Selected board")
 # Use BOARD to search zephyr/boards/** for a _defconfig file,
 # e.g. zephyr/boards/arm/96b_carbon_nrf51/96b_carbon_nrf51_defconfig. When
 # found, use that path to infer the ARCH we are building for.
-find_path(BOARD_DIR NAMES "${BOARD}_defconfig" PATHS ${ZEPHYR_BASE}/boards/*/* NO_DEFAULT_PATH)
+if(NOT BOARD_ROOT)
+  set(BOARD_ROOT ${ZEPHYR_BASE})
+endif()
+
+find_path(BOARD_DIR NAMES "${BOARD}_defconfig" PATHS ${BOARD_ROOT}/boards/*/* NO_DEFAULT_PATH)
```

Comment:
Nice. This should work. All we need to do is replace the remaining references to

${ZEPHYR_BASE}/boards/${ARCH}/${BOARD}/*

with

${BOARD_ROOT}/boards/${ARCH}/${BOARD}/*

and we should be good.

Some grepping shows:

```
cmake/kconfig.cmake
9:#set(BOARD_DEFCONFIG ${PROJECT_SOURCE_DIR}/boards/${ARCH}/${BOARD}/${BOARD}_defconfig)

cmake/dts.cmake
13:set_ifndef(DTS_SOURCE ${PROJECT_SOURCE_DIR}/boards/${ARCH}/${BOARD_FAMILY}/${BOARD_FAMILY}.dts)
88:  set_ifndef(DTS_BOARD_FIXUP_FILE ${PROJECT_SOURCE_DIR}/boards/${ARCH}/${BOARD_FAMILY}/dts.fixup)

```